src/rfac/rfinput.c

```
@@ -0,0 +1,236 @@
+            cur_list[i_cur].weight);
+     i_cur++;
+
+   } /* if != '#' (no comment) */
+
+   /* increment offset in ctr_buffer */
+   offs += (long)len;
+
+  } /* while: scan through control file */
+
+  n_cur = i_cur-1;
+  cur_list[n_cur].group_id = END_OF_GROUP_ID;
```

Cppcheck (2.19.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)